jcl-over-slf4j/src/main/java/org/apache/commons/logging/impl/SLF4JLogFactory.java

```diff
@@ -56,15 +56,15 @@
 
   /**
    * The {@link org.apache.commons.logging.Log}instances that have already been
-   * created, keyed by logger name.
+   * created, keyed by Logger instance.
    */
-  ConcurrentMap<String, Log> loggerMap;
+  ConcurrentMap<Logger, Log> loggerMap;
```

Comment:
That's a good point! Maybe we can control access with Collections.synchronizedMap(WeakHashMap). WDYT?